src/attr/validators.py

```diff
@@ -135,7 +135,12 @@ class _InValidator(object):
     options = attrib()
 
     def __call__(self, inst, attr, value):
-        if value not in self.options:
+        try:
+            in_options = value in self.options
+        except Exception as e:  # e.g. `1 in "abc"`
```

Comment:
I think we want `TypeError`, not `Exception` here.  Other exceptions don't necessarily imply that `value` isn't one of the items in `self.options`.